File: Frontend/src/app/models/commands/throw-command.ts
import { ScriptCommand } from './script-command';
import { ThrowElementType } from './throw-element-type';
import { ThrowType } from './throw-type';

export interface ThrowCommand extends ScriptCommand {
  throwType: ThrowType;
  damage: number;
  knockbackGrowth: number;
  weightDependantKnockback: number;
  throwElement: ThrowElementType;
  angle: number;
  baseKnockback: number;
}

File: backend/FightCore.Models/Subactions/Commands/ThrowCommand.cs
namespace FightCore.Models.Subactions.Commands
{
	public class ThrowCommand : ScriptCommand
	{
		public ThrowType ThrowType { get; set; }

		public int Damage { get; set; }

		public int KnockbackGrowth { get; set; }

		public int WeightDependantKnockback { get; set; }

		public ThrowElementType ThrowElement { get; set; }

		public int Angle { get; set; }

		public int BaseKnockback { get; set; }
	}
}
